src/praisonai-agents/praisonaiagents/hooks/runner.py

```
@@ -147,11 +147,35 @@ def execute_sync(
+            
+            with concurrent.futures.ThreadPoolExecutor(max_workers=1) as pool:
+                try:
+                    # Give hooks reasonable timeout to complete (30s)
+                    future = pool.submit(_run_in_thread)
+                    return future.result(timeout=30)
+                except concurrent.futures.TimeoutError:
+                    # If hooks timeout, log warning and return empty results
+                    # rather than blocking indefinitely
+                    logging.warning(f"Hook execution timeout for event {event}")
+                    return []
+                except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)